infrastructures/kubernetes/src/main/java/org/eclipse/che/workspace/infrastructure/kubernetes/server/secure/SecureServerExposerFactoryProvider.java

```diff
@@ -11,53 +11,67 @@
  */
 package org.eclipse.che.workspace.infrastructure.kubernetes.server.secure;
 
-import static java.util.stream.Collectors.joining;
+import static java.lang.String.format;
 
 import java.util.Map;
 import javax.inject.Inject;
 import javax.inject.Named;
-import javax.inject.Provider;
+import org.eclipse.che.api.workspace.server.model.impl.WarningImpl;
 import org.eclipse.che.inject.ConfigurationException;
 import org.eclipse.che.workspace.infrastructure.kubernetes.environment.KubernetesEnvironment;
 
 /**
  * Provides implementation of {@link SecureServerExposerFactory} according to configuration property
- * with name `che.server.secure_exposer`.
+ * or workspace config attribute with name `che.server.secure_exposer`.
```

Comment:
It makes sense to rewrite this java doc to make clear priority of different configuration sources.

Reply:
fixed